// @amplitude/analytics-react-native/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Identify", {
  enumerable: true,
  get: function () {
    return _analyticsCore.Identify;
  }
});
Object.defineProperty(exports, "Revenue", {
  enumerable: true,
  get: function () {
    return _analyticsCore.Revenue;
  }
});
exports.add = exports.Types = void 0;
Object.defineProperty(exports, "createInstance", {
  enumerable: true,
  get: function () {
    return _reactNativeClient.createInstance;
  }
});
exports.track = exports.setUserId = exports.setSessionId = exports.setOptOut = exports.setGroup = exports.setDeviceId = exports.revenue = exports.reset = exports.remove = exports.logEvent = exports.init = exports.identify = exports.groupIdentify = exports.getUserId = exports.getSessionId = exports.getDeviceId = exports.flush = exports.extendSession = void 0;
var _reactNativeClient = _interopRequireWildcard(require("./react-native-client"));
var _analyticsCore = require("@amplitude/analytics-core");
var Types = _interopRequireWildcard(require("./types"));
exports.Types = Types;
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/* eslint-disable @typescript-eslint/unbound-method */

const {
  add,
  flush,
  getDeviceId,
  getSessionId,
  getUserId,
  groupIdentify,
  identify,
  init,
  logEvent,
  remove,
  reset,
  revenue,
  setDeviceId,
  setGroup,
  setOptOut,
  setSessionId,
  setUserId,
  track,
  extendSession
} = _reactNativeClient.default;

// Export types to maintain backward compatibility with `analytics-types`.
// In the next major version, only export customer-facing types to reduce the public API surface.
exports.extendSession = extendSession;
exports.track = track;
exports.setUserId = setUserId;
exports.setSessionId = setSessionId;
exports.setOptOut = setOptOut;
exports.setGroup = setGroup;
exports.setDeviceId = setDeviceId;
exports.revenue = revenue;
exports.reset = reset;
exports.remove = remove;
exports.logEvent = logEvent;
exports.init = init;
exports.identify = identify;
exports.groupIdentify = groupIdentify;
exports.getUserId = getUserId;
exports.getSessionId = getSessionId;
exports.getDeviceId = getDeviceId;
exports.flush = flush;
exports.add = add;
